Engine/ac/route_finder_impl.cpp

```diff
@@ -218,6 +220,9 @@ void calculate_move_stage(MoveList * mlsp, int index)
   //fixed newxmove = useMoveSpeed * fcos(angl);
   float newxmove = useMoveSpeed * cos(angl);
 
+  // validate that the computed movement isn't larger than the set maxima
+  assert(newxmove <= move_speed_x && newymove <= move_speed_y);
```

Comment:
I think there's a mistake in this assert: newxmove and newymove are signed values, so you might need to take std::abs(newxmove).

Reply:
Oh sorry, I misread where they are getting negated.